applications/RomApplication/custom_strategies/rom_builder_and_solver.h

```diff
@@ -708,6 +708,41 @@ class ROMBuilderAndSolver : public BuilderAndSolver<TSparseSpace, TDenseSpace, T
         KRATOS_CATCH("")
     }
 
+    void CalculateReactions(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb) override
+    {
+        KRATOS_TRY
+        // Get ProcessInfo from main model part
+        const auto& r_current_process_info = rModelPart.GetProcessInfo();
+
+        struct my_tls {
+            Kratos::Vector rhs_elem;
+        };
+
+        auto& r_elements = rModelPart.Elements();
+        if(!r_elements.empty())
+        {
+            block_for_each(r_elements, my_tls(), [&](Element& r_element, my_tls& rTLS)
+            {
+                Element::GeometryType& geom = r_element.GetGeometry();
+                DofsVectorType dofs = {};
+                r_element.CalculateRightHandSide(rTLS.rhs_elem, r_current_process_info);
+                r_element.GetDofList(dofs, r_current_process_info);
+                for (IndexType i = 0; i < dofs.size(); ++i){
+                    if (dofs[i]->IsFixed()){
+                        rb[dofs[i]->EquationId()] += rTLS.rhs_elem[i]; // Building RHS.
+                        dofs[i]->GetSolutionStepReactionValue() = -rb[dofs[i]->EquationId()]; // Reactions are not Zero in the next step for ROMs. Reaction will be overwritten if dof is repeated in other elements, nontheless, it avoids to double loop.
+                    }
+                }
+
+            });
+        }   
+        KRATOS_CATCH("")
+    }

```

Comment:
This is a very delicate class. Please move changes in this file to a separate PR.

Reply:
I agree with your point. It would be necessary to merge the reactions calculation before merging this pull request, as the reactions are needed to properly couple the problem and ensure a valid test.